File: 00-Test/src/main/java/com/leecode/排序_数组/_704_二分查找.java
package com.leecode.排序_数组;

//输入: nums = [-1,0,3,5,9,12], target = 9
//输出: 4
//解释: 9 出现在 nums 中并且下标为 4

import java.util.HashMap;
import java.util.Map;

public class _704_二分查找 {
    public int search2(int[] nums, int target) {
        int left = 0;
        int right = nums.length;
        while(left < right){
            int mid = left + ((right -left)>>1);
            if (nums[mid] > target){
               right = mid;
            }else if  (nums[mid] == target){
                return mid;
            }else if (nums[mid] < target){
                left = mid+1;
            }
        }
        return  -1;
    }





    public int search(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i],i);
        }
        if (map.get(target)!=null){

            return map.get(target);
        }else {
            return  -1;
        }
    }
}

File: 00-Test/src/main/java/com/leecode/高频题/_704_二分查找.js
var search = function (nums, target) {
    let mid, left = 0, right = nums.length - 1;
    while (left <= right) {
        mid = left + ((right - left) >> 1);
        if (target < nums[mid]) {
            right = mid - 1;
        } else if (target > nums[mid]) {
            left = mid + 1;
        } else {
            return mid;
        }
    }
    return -1;
}
let res = search([1, 2, 3, 4, 5, 6, 7], 5)
console.log(res)